File: Daily/sbgo/Leetcode73-矩阵置零.go
package sbgo

func setZeroes(matrix [][]int) {
	m, n := len(matrix), len(matrix[0])
	col, row := 1, 1
	for i := 0; i < m; i++ {
		if matrix[i][0] == 0 {
			col = 0
		}
	}
	for i := 0; i < n; i++ {
		if matrix[0][i] == 0 {
			row = 0
		}
	}
	for i := 1; i < m; i++ {
		for j := 1; j < n; j++ {
			if matrix[i][j] == 0 {
				matrix[i][0] = 0
				matrix[0][j] = 0
			}
		}
	}
	for i := 1; i < m; i++ {
		for j := 1; j < n; j++ {
			if matrix[i][0] == 0 || matrix[0][j] == 0 {
				matrix[i][j] = 0
			}
		}
	}
	if col == 0 {
		for i := 0; i < m; i++ {
			matrix[i][0] = 0
		}
	}
	if row == 0 {
		for i := 0; i < n; i++ {
			matrix[0][i] = 0
		}
	}
}

File: Leetcode/PY/Leetcode73-矩阵置零.py
from typing import List
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        if not matrix:
            return 
        m,n = len(matrix),len(matrix[0])
        col,row = 1,1
        for i in range(m):
            if matrix[i][0] == 0:
                col = 0
        for i in range(n):
            if matrix[0][i] == 0:
                row = 0
        for i in range(1,m):
            for j in range(1,n):
                if matrix[i][j] == 0:
                    matrix[i][0] = matrix[0][j] = 0
        for i in range(1,m):
            for j in range(1,n):
                if matrix[i][0] == 0 or matrix[0][j] == 0:
                    matrix[i][j] = 0
        if col == 0:
            for i in range(m):
                matrix[i][0] = 0

        if row == 0:
            for i in range(n):
                matrix[0][i] = 0